src/server/search/search_family.cc

```diff
@@ -48,48 +48,30 @@ search::SchemaField::VectorParams ParseVectorParams(CmdArgParser* parser) {
   search::SchemaField::VectorParams params{};
 
   params.use_hnsw = parser->Switch("HNSW", true, "FLAT", false);
-  size_t num_args = parser->Next<size_t>();
+  const size_t num_args = parser->Next<size_t>();
 
   for (size_t i = 0; i * 2 < num_args; i++) {
-    if (parser->Check("DIM").ExpectTail(1)) {
+    if (parser->Check("DIM")) {
       params.dim = parser->Next<size_t>();
-      continue;
-    }
-
-    if (parser->Check("DISTANCE_METRIC").ExpectTail(1)) {
+    } else if (parser->Check("DISTANCE_METRIC")) {
       params.sim = parser->Switch("L2", search::VectorSimilarity::L2, "COSINE",
                                   search::VectorSimilarity::COSINE);
-      continue;
-    }
-
-    if (parser->Check("INITIAL_CAP").ExpectTail(1)) {
-      params.capacity = parser->Next<size_t>();
-      continue;
-    }
-
-    if (parser->Check("M").ExpectTail(1)) {
+    } else if (parser->Check("INITIAL_CAP")) {
+      parser->Next<size_t>();
+      // params.capacity = parser->Next<size_t>();
+    } else if (parser->Check("M")) {
       params.hnsw_m = parser->Next<size_t>();
-      continue;
-    }
-
-    if (parser->Check("EF_CONSTRUCTION").ExpectTail(1)) {
+    } else if (parser->Check("EF_CONSTRUCTION")) {
       params.hnsw_ef_construction = parser->Next<size_t>();
-      continue;
-    }
-
-    if (parser->Check("EF_RUNTIME").ExpectTail(1)) {
+    } else if (parser->Check("EF_RUNTIME")) {
       parser->Next<size_t>();
       LOG(WARNING) << "EF_RUNTIME not supported";
-      continue;
-    }
-
-    if (parser->Check("EPSILON").ExpectTail(1)) {
+    } else if (parser->Check("EPSILON")) {
       parser->Next<double>();
       LOG(WARNING) << "EPSILON not supported";
-      continue;
+    } else {
+      parser->Skip(2);
```

Comment:
is this just ignoring unsupported flags?

Reply:
yes

Reply:
why dont we send unsupported command?

Reply:
I don't know I haven't changed the logic, just remove redundant code

Reply:
yes I see you did not change the logic here, I am asking to understand
@dranikpg can you answer this?